### PR TITLE
Transaction fees & KPIs

### PR DESCRIPTION
**Pull request by ncomerci**
**Do not merge until transaction tags PR is merged.**

- KPIs migrated to code-based computing
- Out transaction fees analysis
- Code refactors

### CODE REVIEW

**Review by 1emu — APPROVED** [review]
Great work Nico 💪 ! 
I left some suggestions and a PR to add some tests.
I'm approving assuming we address those before merging :)

File: src/export-balances.ts
Comment (andyesp) [inline src/export-balances.ts:52]: You can write this `const balances = flattenArray(...).filter(...)`

File: src/entities/Networks.ts
Comment (andyesp) [inline src/entities/Networks.ts:33]: I'd change this to an object export to keep the same pattern as the other entities.

File: src/entities/Tags.ts
Comment (andyesp) [reply src/entities/Tags.ts:248]: Or just `getCategory`?

File: src/utils.ts
Comment (andyesp) [inline src/utils.ts:28]: You are modifying the original array here. You can copy it and store it in a new variable.
Maybe doing something like `const sortedArray = [...array].sort(...)`

File: src/export-balances.ts
Comment (ncomerci) [reply src/export-balances.ts:35]: There's always a value!

image

File: src/export-kpis.ts
Comment (ncomerci) [reply src/export-kpis.ts:115]: Yes, it's the place for the members percentage. I don't think it makes much sense to set 100% there

image

**Review by andyesp — APPROVED** [review]

File: src/export-transactions.ts
Comment (1emu) [inline src/export-transactions.ts:58]:
```typescript 
function baseCovalentUrl(network:Network){
   return `https://api.covalenthq.com/v1/${network.id}`
}
```

File: src/export-transactions.ts
Comment (1emu) [inline src/export-transactions.ts:77]: I know it's a high number, but eventually we are going to need to check that there are no more records in the next page, right?

If that's the case I'm ok with creating the ticket and merging as is.

Comment (1emu) [reply src/export-transactions.ts:77]: maybe use something like Governance's `inBatches`

**Review by 1emu — APPROVED** [review]
🔥

File: src/export-transactions.ts
Comment (ncomerci) [reply src/export-transactions.ts:77]: Glad you mentioned it, I've already included it in my next PR!

image
image

**Pull request merged**
